src/components/card/Card.jsx

```diff
@@ -0,0 +1,81 @@
+import "./card.css";
+import { useNavigate } from "react-router-dom";
+import { useAuth, useWishList } from "../../context";
+export function Card({ product }) {
+  const { user } = useAuth();
+  const navigate = useNavigate();
+  const { wishList, addToWishList, removeFromWishList } = useWishList();
+
+  const productInWishList = wishList?.some((p) => p._id === product._id)
```

Comment:
Some returns true and false why do you need a ternary operator

Reply:
its optional chaining https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

it will prevent unwanted errors like cannot read the properties of undefined